query-planner-js/src/buildPlan.ts

```diff
@@ -1390,6 +1397,18 @@ function createNewFetchSelectionContext(type: CompositeType, selections: Selecti
   return [inputSelection, path];
 }
 
+function extractPathInParentForKeyFetch(type: CompositeType, path: OperationPath): OperationPath {
+  // A "key fetch" (calls to the `_entities` operation) always have to start with some type-cast into
+  // the entity fetched (`type` in this function), so we can remove a type-cast into the entity from
+  // the parent path if it is the last thing in the past. And doing that removal ensures the code
+  // later reuse fetch groups for different entities, as long as they get otherwise merged into the
+  // parent at the same place.
+  const lastElement = path[path.length - 1];
+  return lastElement && lastElement.kind === 'FragmentElement' && lastElement.typeCondition?.name === type.name
```

Comment:
Could you put parenthesis around the ternary clause so it's a little more readable?

Reply:
Added, thanks.